### PR TITLE
Add smoke test for appprotect csrf policy

### PR DESCRIPTION
**Pull request by vepatel**
### Proposed changes
- Add smoke test for AP CSRF policy,
- Remove hardcoded waits.

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that all unit tests pass after adding my changes
- [ ] I have updated necessary documentation
- [x] I have rebased my branch onto master
- [x] I will ensure my PR is targeting the master branch and pulling from my branch from my own fork

```suite/test_app_protect.py::TestAppProtect::test_responses_csrf[backend_setup0-crd_ingress_controller_with_ap0] PASSED [ 21%]```

### CODE REVIEW

File: tests/suite/test_app_protect.py [hunk 1]
```diff
@@ -196,18 +199,50 @@ def test_responses_malformed_block(
...
+        print("------------- Run test for AP policy: CSRF --------------")
+        print(f"Request URL without CSRF protection: {backend_setup.req_url}")
+        print(f"Request URL with CSRF protection: {backend_setup.req_url_2}")
+
+        ensure_response_from_backend(backend_setup.req_url_2, backend_setup.ingress_host, check404=True)
+
+        print("----------------------- Send request with http origin header ----------------------")
+
+        headers = {"host": backend_setup.ingress_host, "Origin": "http://appprotect.example.com"}
+        resp_valid = requests.post(backend_setup.req_url, headers=headers, verify=False, cookies={"flavor": "darkchoco"})
```
Comment (pleshakov) [inline tests/suite/test_app_protect.py:237]: are cookies necessary? the policy doesn't configure any rules for cookies, right?

File: tests/suite/resources_utils.py [hunk 2]
```diff
@@ -1174,10 +1174,21 @@ def ensure_response_from_backend(req_url, host, additional_headers=None) -> None
...
         headers.update(additional_headers)
-    for _ in range(30):
-        resp = requests.get(req_url, headers=headers, verify=False)
-        if resp.status_code != 502 and resp.status_code != 504:
-            print(f"After {_ * 2} seconds got non 502|504 response. Continue with tests...")
-            return
-        time.sleep(1)
-    pytest.fail(f"Keep getting 502|504 from {req_url} after 60 seconds. Exiting...")
+    
+    if check404:
```
Comment (pleshakov) [inline tests/suite/resources_utils.py:1178]: Woudl it make sense to put the status checks in a function and thus remove if...else completely? something like that

```
for _ in range(60):
            resp = requests.get(req_url, headers=headers, verify=False)
            if  checkRespStatus(check404):
                print(f"After {_} retries at 1 second interval, got {resp.status_code} response. Continue with tests...")
                return
            time.sleep(1)
        pytest.fail(f"Keep getting {resp.status_code} from {req_url} after 60 seconds. Exiting...")
```

**Review by pleshakov — APPROVED** [review]

File: tests/suite/test_app_protect.py [hunk 3]
```diff
@@ -144,15 +147,16 @@ def test_responses_dataguard_alarm(
...
         """
         Test file-block AppProtect policy: Block executing types e.g. .bat and .exe
-        """   
+        """
         print("------------- Run test for AP policy: file-block --------------")
         print(f"Request URL: {backend_setup.req_url} and Host: {backend_setup.ingress_host}")
 
-        wait_before_test(40)
-        ensure_response_from_backend(backend_setup.req_url, backend_setup.ingress_host)
+        ensure_response_from_backend(backend_setup.req_url, backend_setup.ingress_host, check404=True)
```
Comment (soneillf5) [inline tests/suite/test_app_protect.py:159]: it's nice to see the wait removed :D

**Review by soneillf5 — APPROVED** [review]

File: tests/suite/test_app_protect.py (hunk 1 above)
Comment (vepatel) [reply tests/suite/test_app_protect.py:237]: @pleshakov cookies are necessary for CSRF protection to kick in

**Pull request merged**